packages/gatsby-source-wordpress/src/fetch.js

```diff
@@ -136,19 +137,30 @@ Mama Route URL: ${url}
     },
   ]
 
-  if (allRoutes) {
-    let validRoutes = getValidRoutes({
-      allRoutes,
-      url,
-      _verbose,
-      _useACF,
-      _hostingWPCOM,
-      _acfOptionPageIds,
-      _includedRoutes,
-      _excludedRoutes,
-      typePrefix,
-      refactoredEntityTypes,
-    })
+  const typeBuilder = buildRouteType(refactoredEntityTypes)
+  const urlBuilder = urlBuilderFactory(url, _hostingWPCOM)
+  let validRoutes = _buildRoutesWithParams(typeBuilder, pathBuilder, urlBuilder)
```

Comment:
let's do 
```suggestion
  let validRoutes = await _buildRoutesWithParams(typeBuilder, pathBuilder, urlBuilder)
```

to enable this function to be async